src/log.h

```diff
@@ -24,44 +25,51 @@ extern "C" {
 #define LOG_D(tag, message, ...) \
     log_message(tag, LOG_LEVEL_DEBUG, message, __VA_ARGS__)
 
+#define LOG_PRODUCER_DEFAULT PRODUCER
+
+#define LOG_PRODUCER_CREATE_LOCAL(TAG, SUFFIX, VAR) \
+    static log_producer_t* VAR; \
+    FUNCTION_CTOR(concat(log_producer_local_init, SUFFIX), { \
+        VAR = log_producer_init(TAG); \
+    }) \
+    FUNCTION_DTOR(concat(log_producer_local_free, SUFFIX), { \
+        log_producer_free(VAR); \
+    })
+
+#define LOG_PRODUCER_CREATE_LOCAL_DEFAULT(TAG, SUFFIX) \
+    LOG_PRODUCER_CREATE_LOCAL(TAG, SUFFIX, LOG_PRODUCER_DEFAULT)
+
 enum log_level {
-    LOG_LEVEL_ERROR,
-    LOG_LEVEL_RECOVERABLE,
-    LOG_LEVEL_WARNING,
-    LOG_LEVEL_INFO,
-    LOG_LEVEL_VERBOSE,
-    LOG_LEVEL_DEBUG,
-    LOG_LEVEL_DEBUG_INTERNALS,
+    LOG_LEVEL_ERROR = 1u << 6u,
```

Comment:
It would be more readable when bit shifts would be replaced with constants in hex or decimal form.

Reply:
Fixed, thanks!